cstar/tests/roms/fixtures.py

```diff
@@ -0,0 +1,155 @@
+        pup_test_data.fetch("TPXO_global_test_data.nc")
+        pup_test_data.fetch("ERA5_NA_2012.nc")
+        pup_test_data.fetch("CESM_BGC_2012.nc")
+        pup_test_data.fetch("CESM_BGC_SURFACE_2012.nc")
+
+        if symlink_path.is_symlink():
+            raise FileExistsError(
+                f"{symlink_path} already exists as a symlink to {symlink_path.resolve()}"
+            )
+
+        symlink_path.symlink_to(ROMS_TOOLS_DATA_DIRECTORY)
+
+        def cleanup():
+            if symlink_path.is_symlink():
+                print(f"removing {symlink_path}")
+                symlink_path.unlink()
+
+        request.addfinalizer(cleanup)
+
+    return _fetch_roms_tools_source_data
+
+
+@pytest.fixture
+def fetch_remote_test_case_data() -> Callable[[], None]:
+    """
+    Fixture that provides a function to fetch remote test case data from a GitHub repository.
+
+    Returns
+    -------
+    Callable[[], None]
+        A factory function that fetches the remote test case data and sets it up in the local
+        test data directory.
+    """
+
+    def _fetch_remote_test_case_data() -> None:
+        """
+        Downloads and sets up the remote test case data for testing.
+
+        This function downloads a zip archive of a specific commit of a repo containing test data,
+        extracts the archive into the specified test data directory, and performs cleanup of
+        intermediate files and directories.
+
+        Data are saved to the CSTAR_TEST_DATA_DIRECTORY, set in tests/config.py
+
+        Returns
+        -------
+        None
+        """
+
+        test_case_repo_url = (
+            "https://github.com/CWorthy-ocean/cstar_blueprint_test_case/"
+        )
+        checkout_target = "d7996aea8d4fd4ca4148b34d2d898f019c90a8ff"
+
+        # Construct the URL of this commit as a zip archive:
+        archive_url = f"{test_case_repo_url.rstrip('/')}/archive/{checkout_target}.zip"
+
+        # Create a cache dir with pooch
+        # cache_dir = Path(pooch.os_cache("cstar_test_case_data"))
+
+        # Download the zip with pooch
+        zip_path = pooch.retrieve(
+            url=archive_url,
+            known_hash="e291bc79dd5f0ab88f486974075f3defe2076ff3706787131dae7fd0f01358b5",
+            fname=f"{checkout_target}.zip",  # Name of the cached file
+            path=CSTAR_TEST_DATA_DIRECTORY,  # Set the cache directory (customize as needed)
+        )
+
+        # Unzip the files into a subdirectory `extract_dir` of `cache_dir`
+        with zipfile.ZipFile(zip_path, "r") as zip_ref:
+            extract_list = zip_ref.namelist()
+            extract_dir = CSTAR_TEST_DATA_DIRECTORY / extract_list[0]
+            zip_ref.extractall(CSTAR_TEST_DATA_DIRECTORY)
+
+        # Copy the contents of the subdirectory up one and remove it and the zip file
+        shutil.copytree(extract_dir, CSTAR_TEST_DATA_DIRECTORY, dirs_exist_ok=True)
+        shutil.rmtree(extract_dir)
+        Path(zip_path).unlink()
```

Comment:
Why is this moving around of files on-disk necessary?

Reply:
All the test data is in a git repo. This routine downloads the repo as a zip archive, rather than cloning it with all its history and baggage. Unfortunately it can only be unzipped to a subdirectory of where we want it to go, so this moves the contents of the subdirectory up one level.

Reply:
Huh. You can't just unzip into the `tmpdir`?

Reply:
not directly, it would go into a named subdirectory of wherever you ask it to be unzipped. The name isn't specifiable either, it's written into the zip file metadata. Kind of annoying but livable imho.

Reply:
That is a bit annoying, but yeah not the end of the world.